models-info/src/main/java/com/flipkart/fdp/ml/transformer/PopularWordsEstimatorTransformer.java

```diff
@@ -0,0 +1,45 @@
+package com.flipkart.fdp.ml.transformer;
+
+import com.flipkart.fdp.ml.modelinfo.PopularWordsEstimatorModelInfo;
+
+import java.util.*;
+
+public class PopularWordsEstimatorTransformer implements Transformer {
+	private PopularWordsEstimatorModelInfo modelInfo;
+
+	public PopularWordsEstimatorTransformer(final PopularWordsEstimatorModelInfo modelInfo) {
+		this.modelInfo = modelInfo;
+	}
+
+	public double predict(final String[] words) {
+		List<String> matchedWords = getMatchedWords(modelInfo.getPopularWords(), words);
+		return matchedWords.size() * 1.0 / words.length;
```

Comment:
getMatchedWords is private, so I guess we don't need actual list of matched words. Hence, just the count of matched words should be good and performant for prediction.

Reply:
changed